## Setup

In [ ]:
import pandas as pd
pd.options.display.float_format = '{:.1f}'.format

In [ ]:
ufos_path = 'YOUR PATH HERE'
sightings = pd.read_csv(ufos_path)
sightings.head()

<ipython-input-2-828416dd11ab>:2: DtypeWarning: Columns (5,9) have mixed types. Specify dtype option on import or set low_memory=False.
  sightings = pd.read_csv(ufos_path)


datetime                  city state country     shape  \
0  10/10/1949 20:30            san marcos    tx      us  cylinder   
1  10/10/1949 21:00          lackland afb    tx     NaN     light   
2  10/10/1955 17:00  chester (uk/england)   NaN      gb    circle   
3  10/10/1956 21:00                  edna    tx      us    circle   
4  10/10/1960 20:00               kaneohe    hi      us     light   

  duration (seconds) duration (hours/min)  \
0               2700           45 minutes   
1               7200              1-2 hrs   
2                 20           20 seconds   
3                 20             1/2 hour   
4                900           15 minutes   

                                            comments date posted    latitude  \
0  This event took place in early fall around 194...   4/27/2004  29.8830556   
1  1949 Lackland AFB&#44 TX.  Lights racing acros...  12/16/2005    29.38421   
2  Green/Orange circular disc over Chester&#44 En...   1/21/2008        53.2   
3  My older brother and twin sister were leaving ...   1/17/2004  28.9783333   
4  AS a Marine 1st Lt. flying an FJ4B fighter/att...   1/22/2004  21.4180556   

   longitude  
0      -97.9  
1      -98.6  
2       -2.9  
3      -96.6  
4     -157.8

## How does the average duration of sightings in the United States compare to the average duration of sightings everywhere else?

In [ ]:
sightings['duration (seconds)'] = pd.to_numeric(sightings['duration (seconds)'], errors='coerce')

In [ ]:
us_sightings = sightings[sightings['country'] == 'us']
us_sightings['duration (seconds)'].mean() / 60

96.66986747962484

In [ ]:
non_us_sightings = sightings[sightings['country'] != 'us']
non_us_sightings['duration (seconds)'].mean() / 60

379.70976898753594

**Average duration of US sightings: 96.7 minutes**

**Average duration of non-US sightings: 379.7 minutes**

## What is the most common/frequent sighting "shape" among sightings above 66.5 degrees latitude?

In [ ]:
sightings['latitude'] = pd.to_numeric(sightings['latitude'], errors='coerce')

In [ ]:
arctic_sightings = sightings[sightings['latitude'] > 66.5]
arctic_sightings.describe(include='all')

datetime              city state country  shape  \
count                18                18    12       7     17   
unique               17                15     2       2      7   
top     8/13/2007 00:45  tromsoe (norway)    ak      us  light   
freq                  2                 3     8       4      6   
mean                NaN               NaN   NaN     NaN    NaN   
std                 NaN               NaN   NaN     NaN    NaN   
min                 NaN               NaN   NaN     NaN    NaN   
25%                 NaN               NaN   NaN     NaN    NaN   
50%                 NaN               NaN   NaN     NaN    NaN   
75%                 NaN               NaN   NaN     NaN    NaN   
max                 NaN               NaN   NaN     NaN    NaN   

        duration (seconds) duration (hours/min)                    comments  \
count                 18.0                   18                          18   
unique                 NaN                   18                          18   
top                    NaN             10-15min  Radar sighting and filming   
freq                   NaN                    1                           1   
mean                 498.8                  NaN                         NaN   
std                  724.1                  NaN                         NaN   
min                    5.0                  NaN                         NaN   
25%                   45.0                  NaN                         NaN   
50%                  240.0                  NaN                         NaN   
75%                  750.0                  NaN                         NaN   
max                 2700.0                  NaN                         NaN   

       date posted  latitude  longitude  
count           18      18.0       18.0  
unique          17       NaN        NaN  
top      10/8/2007       NaN        NaN  
freq             2       NaN        NaN  
mean           NaN      69.7      -86.4  
std            NaN       1.8       79.3  
min            NaN      66.9     -162.6  
25%            NaN      67.9     -154.6  
50%            NaN      69.8     -134.0  
75%            NaN      70.6       15.2  
max            NaN      72.7       19.0

**light**

## What is the median latitude among sightings less than or equal to 10 seconds in duration?

In [ ]:
short_sightings = sightings[sightings['duration (seconds)'] <= 10]
short_sightings['latitude'].median()

39.46438585

**39.5 degrees**